src/main/scala/akka/persistence/jdbc/journal/dao/ByteArrayJournalDao.scala

```diff
@@ -86,9 +84,14 @@ trait BaseByteArrayJournalDao extends JournalDaoWithUpdates {
     }
   }
 
-  private def writeJournalRows(xs: Seq[JournalRow]): Future[Unit] = for {
-    _ <- db.run(queries.writeJournalRows(xs))
-  } yield ()
+  private def writeJournalRows(xs: Seq[JournalRow]): Future[Unit] = {
+    if (slickProfile != H2Profile)
+      // Write atomically without auto-commit
+      db.run(queries.writeJournalRows(xs).transactionally).map(_ => Unit)
+    else
+      // However transactionally causes H2 tests to fail
+      db.run(queries.writeJournalRows(xs)).map(_ => Unit)
```

Comment:
Why H2 is failing when using `transactionally`? That's not normal.

Reply:
I forgot I had added this comment and merge it. 

In any case, I don't think we are doing any harm here. Actually, I run this locally without the if/else and calling `transactionally`. All H2 tests passed. 

Moreover, we don't need it transactionally because we have only one `DBIO`. `transactionally` is only needed when we have many DBIOs and we want to run them all together in the same tx. That's not the case here. Here we have one `DBIO` with many row inserts.

Reply:
Yes the test failures were intermittent. Locally I also didn't have a problem, but maybe the Travis hardware was lower spec and caused the circuit breaker to time out. It's also not clear why this would reduce performance in H2 in the first place.

With regards to not requiring `transactionally`, it certainly made a significant difference to SQL Server performance. It's calling `JournalTableC ++= xs.sortBy(_.sequenceNumber)` but I have no idea how Slick handles this internally.